plugins/out_kinesis_streams/kinesis.c

```diff
@@ -108,6 +108,9 @@ static int cb_kinesis_init(struct flb_output_instance *ins,
     if (tmp) {
         ctx->custom_endpoint = FLB_TRUE;
         ctx->endpoint = removeProtocol((char *) tmp, "https://");
+        if (ctx->endpoint[strlen(ctx->endpoint)-1] == '/') {
+            ctx->endpoint[strlen(ctx->endpoint)-1] = '\0';
+        }
```

Comment:
So this task is potentially useful for all AWS plugins, would you be willing to make it a modification to the above `removeProtocol` function? and change its name to something like `prepareEndpoint` (also please try to think of a better name, I am not happy with that idea that my mind just came up with ;).

Reply:
Hello Wesley Pettit,
sorry i've missed communication due to old company email in github preferences.
Yep, it's a common problem for kinesis streams and firehouse and cloudwatch logs as well.
Not sure we should rework removeProtocol function, because the function perform simple and clear action and also have an argument. Can propose to create function removeTrailSlash and use it where required.

Checked function flb_utils_url_split but  it's not suitable, because protocol which is not required by kinesis plugin documentation.
Checked function replace_uri_tokens from flb_aws_utils, no luck can't be used to remove characters.

Can implement approach with removeTrailSlash, if you have a better idea, please share.
Thank you!

Reply:
Hello @PettitWesley,
just prepared code changes.
Updated tests as well.
Thank you!